lib/hooks/validate/status.js

```diff
@@ -1,14 +1,27 @@
-const flow = require('../../flow');
 const BadRequestError = require('../../errors/bad-request');
+const getNextStepsForUser = require('../../flow/get-next-steps-for-user');
+const { withNtco, withLicensing } = require('../../flow/status');
 
 module.exports = () => {
   return model => {
-    const validNextSteps = flow.getNextSteps(model.meta.previous).map(step => step.id);
+    const currentStatus = model.status;
+    const nextStatus = model.meta.next;
 
-    if (!validNextSteps.includes(model.meta.next)) {
-      throw new BadRequestError('Invalid status change');
-    }
+    return Promise.resolve()
+      .then(() => {
+        // resubmitting an application triggers an auto-status change, don't need to validate against the user
+        if (currentStatus === 'resubmitted' && [withNtco.id, withLicensing.id].includes(nextStatus)) {
+          return;
+        }
```

Comment:
I'd just add these to the available next steps for the status of `resubmitted` in the same way as every other status and allow the logic to be the same here.

This is doing the exact same thing as every other status, but in a _special_ way, for no reason that I can see.

Reply:
Actually, now I possibly see - is this because once it _has_ a status of resubmitted then it no longer appears in the user's "outstanding" list, and so validating the ability of the user to modify its status fails?